### ISSUE TITLE
(Feature Request) Sphinx/reStructuredText Support

### ISSUE THREAD

**Issue by ShaneKosieradzki** [hdoc/hdoc#3]
I like this project's approach to documentation whereby you are leveraging the LLVM compiler to associate doc-comments with elements of the AST. This approach is great as it will allow the documentation tooling to closely follow the C/C++ language standard. However, the full potential of such an association is not being harnessed. 

While the auto-generated output is convenient for most applications, it can be suffocating when the user is trying to create more custom output. [Sphinx](https://www.sphinx-doc.org/en/master/index.html)—the project used to document Python—has solved this problem by using a more elaborate form of Markdown known as [reStructuredText](https://docutils.sourceforge.io/rst.html) that supports user-definable subroutines, as well as language-specific [Domains](https://www.sphinx-doc.org/en/master/usage/restructuredtext/domains.html?highlight=domain).  By default Sphinx defines the [PythonDomain](https://www.sphinx-doc.org/en/master/usage/restructuredtext/domains.html?highlight=domain#the-python-domain), [CDomain](https://www.sphinx-doc.org/en/master/usage/restructuredtext/domains.html?highlight=domain#the-c-domain), and the [C++Domain](https://www.sphinx-doc.org/en/master/usage/restructuredtext/domains.html?highlight=domain#cpp-domain). 

You'll notice that in each domain there are various language-specific entity tags such as `:py:mod:` for a Python Module, or `:cpp:enum:` for a C++ Enumeration.  These tags can be used in the documentation source to create cross-references to said entity or make associations between them. Currently, tags from the Python Domain are auto-parsed from the source by Sphinx, but `C/C++` entities have to be specifically declared in the documentation source using a pattern like this `.. cpp:enumerator:: name`.

This project should leverage its tight coupling with LLVM to expose a Sphinx-Facing API/Plugin that will auto-parse C/C++ source files and load documented entities into the Sphinx Environment. If this project is open for collaboration I would be happy to share my Python/Sphinx experience with the team.

**Comment by hdoc** [comment]
Thank you for your kind offer. We would be willing to look at PRs for such a feature, but this is not something our team will be working on in the near term unless there is a customer need for it. If you work at a company that needs this feature, please reach out to us at [contact@hdoc.io](mailto:contact@hdoc.io).

I agree that reStructuredText is the better format for programming documentation and we have an issue in our internal tracker for implementing RST support in hdoc. However, Sphinx support is not currently a priority, for the following reasons:
* We already generate HTML documentation that is of sufficient quality for most use cases, i.e. it's reached the 80% milestone. The following 20% to implement customization and other features is reaching diminishing returns territory.
* hdoc is designed to be a self-contained tool, and the complex multi-stage pipelines that characterize some of the Doxygen-Exhale-Sphinx workflows are exactly what we set out to replace. [Evidently we are not the only ones that feel this way](https://medium.com/practical-coding/c-documentation-with-doxygen-cmake-sphinx-breathe-for-those-of-use-who-are-totally-lost-7d555386fe13). Our single-binary approach simplifies deployment, and adding Sphinx as an output would reduce that advantage.
* There are other features that have not yet reached that 80% milestone, and our priority is to develop those features first. This calculus changes if a paying customer is requesting a feature.

In summary, Sphinx has some compelling features but integration with hdoc at this at this stage in hdoc's development is not a priority for our team. Sorry.
